src/os/bluestore/BlueStore.cc

```diff
@@ -5633,7 +5635,13 @@ int BlueStore::_do_write_zero(
   uint64_t length)
 {
   bufferlist zl;
-  zl.append_zero(length);
+  uint64_t len = length;
+  while (len > 0) {
+    bufferlist t;
+    t.append(zeros, 0, MIN(zeros.length(), len));
+    t len -= t.length();
+    zl.claim_append(t);
```

Comment:
Instead of that, you could just create a bunch of bufferptrs and append them to zl directly - no need for interim bufferlist (its construction is quite expensive).